File: packages/react-native-vision-camera-location/ios/HybridLocationManagerFactory.swift
///
/// HybridLocationManager.swift
/// VisionCameraLocation
/// Copyright © 2025 Marc Rousavy @ Margelo
///

import CoreLocation
import VisionCamera

final class HybridLocationManagerFactory: HybridLocationManagerFactorySpec {
  func createLocationManager(options: LocationManagerOptions) -> any HybridLocationManagerSpec {
    return HybridLocationManager(options: options)
  }

  func createLocation(latitude: Double, longitude: Double) -> any HybridLocationSpec {
    let location = CLLocation(latitude: latitude, longitude: longitude)
    return HybridLocation(location: location)
  }
}

File: packages/react-native-vision-camera-location/android/src/main/java/com/margelo/nitro/camera/location/HybridLocationManagerFactory.kt
package com.margelo.nitro.camera.location

import android.location.Location
import androidx.annotation.Keep
import com.facebook.jni.annotations.DoNotStrip
import com.margelo.nitro.camera.HybridLocationSpec

@DoNotStrip
@Keep
class HybridLocationManagerFactory : HybridLocationManagerFactorySpec() {
  @DoNotStrip
  @Keep
  override fun createLocationManager(options: LocationManagerOptions): HybridLocationManagerSpec {
    return HybridLocationManager(options)
  }

  override fun createLocation(
    latitude: Double,
    longitude: Double,
  ): HybridLocationSpec {
    val location =
      Location("custom_created").apply {
        this.latitude = latitude
        this.longitude = longitude
      }
    return HybridLocation(location)
  }
}
